pkg/deployments/tasks/20220823-arbitrum-root-gauge-factory-v2/test/task.fork.ts

```diff
@@ -0,0 +1,259 @@
+      periodTime: firstMintWeekTimestamp.sub(WEEK), // Process past week, which had zero votes
+      periodEmissions: 0,
+    });
+    // No token transfers are performed if the emissions are zero, but we can't test for a lack of those
+
+    await advanceTime(WEEK);
+
+    // The gauge should now mint and send all minted tokens to the Arbitrum bridge
+    const mintTx = await authorizerAdaptor.connect(admin).performAction(gauge.address, calldata, { value: bridgeETH });
+    const event = expectEvent.inIndirectReceipt(await mintTx.wait(), gauge.interface, 'Checkpoint', {
+      periodTime: firstMintWeekTimestamp,
+    });
+    const actualEmissions = event.args.periodEmissions;
+
+    // The amount of tokens minted should equal the weekly emissions rate times the relative weight of the gauge
+    const weeklyRate = (await BALTokenAdmin.getInflationRate()).mul(WEEK);
+
+    const expectedEmissions = weightCap.mul(weeklyRate).div(FP_SCALING_FACTOR);
+    expectEqualWithError(actualEmissions, expectedEmissions, 0.001);
+
+    // Tokens are minted for the gauge
+    expectTransferEvent(
+      await mintTx.wait(),
+      {
+        from: ZERO_ADDRESS,
+        to: gauge.address,
+        value: actualEmissions,
+      },
+      BAL
+    );
+
+    // And the gauge then deposits those in the predicate via the bridge mechanism
+    const bridgeInterface = new ethers.utils.Interface([
+      'event DepositInitiated(address l1Token, address indexed from, address indexed to, uint256 indexed sequenceNumber, uint256 amount)',
+    ]);
+
+    expectEvent.inIndirectReceipt(await mintTx.wait(), bridgeInterface, 'DepositInitiated', {
+      from: gauge.address,
+      to: recipient.address,
+      l1Token: BAL,
+      amount: actualEmissions,
+    });
+  });
+
+  it('mint multiple weeks', async () => {
+    const numberOfWeeks = 5;
+    await advanceTime(WEEK * numberOfWeeks);
+    await gaugeController.checkpoint_gauge(gauge.address);
+
+    const weekTimestamp = await currentWeekTimestamp();
+
+    // We can query the relative weight of the gauge for each of the weeks that have passed
+    const relativeWeights: BigNumber[] = await Promise.all(
+      range(1, numberOfWeeks + 1).map(async (weekIndex) =>
+        gaugeController['gauge_relative_weight(address,uint256)'](gauge.address, weekTimestamp.sub(WEEK * weekIndex))
+      )
+    );
+
+    // We require that they're all above the cap for simplicity - this lets us use the cap as each week's weight (and
+    // also tests cap behavior).
+    for (const relativeWeight of relativeWeights) {
+      expect(relativeWeight).to.be.gt(weightCap);
+    }
+
+    // The amount of tokens allocated to the gauge should equal the sum of the weekly emissions rate times the weight
+    // cap.
+    const weeklyRate = (await BALTokenAdmin.getInflationRate()).mul(WEEK);
+    const expectedEmissions = weightCap.mul(numberOfWeeks).mul(weeklyRate).div(FP_SCALING_FACTOR);
```

Comment:
```suggestion
    // Note that instead of the weight, we use the cap (since we expect for the weight to be larger than the cap)
    const expectedEmissions = weightCap.mul(numberOfWeeks).mul(weeklyRate).div(FP_SCALING_FACTOR);
```